cloud/endagaweb/views/towers.py

```diff
@@ -28,12 +32,17 @@
 from endagaweb import models
 from endagaweb.views.dashboard import ProtectedView
 from endagaweb.views import django_tables
+from django.template.loader import get_template
+from django.http import HttpResponse
 
 
-class TowerList(drf_views.APIView):
+class TowerList(PermissionRequiredMixin, drf_views.APIView):
```

Comment:
should check against the network instance:

> If a get_object() method is defined either manually or by including another mixin (for example SingleObjectMixin) or self.object is defined then the permission will be tested against that specific instance, alternatively you can specify get_permission_object() method if self.object or get_object() does not return the object against you want to test permission

http://django-guardian.readthedocs.io/en/stable/api/guardian.mixins.html#guardian.mixins.PermissionRequiredMixin